nntrainer/models/neuralnet.cpp

```diff
@@ -381,18 +385,22 @@ void NeuralNetwork::load(const std::string &file_path,
       }
       loadedWeight = true;
       bin_file_pos = model_file.tellg();
+      load_path = file_path;
+      return;
     }
     try {
       /// this is assuming that the failure is allowed at the end of the file
       /// read. so, after this line, additional read shouldn't be called
       model_file.seekg(bin_file_pos);
 
-      std::string opt_type;
-      opt_type = readString(model_file);
-      if (istrequal(opt_type, "adam") && istrequal(opt->getType(), "adam")) {
-        for (auto iter = model_graph.cbegin(); iter != model_graph.cend();
-             iter++) {
-          (*iter)->read(model_file, true);
+      if (istrequal(opt->getType(), "adam")) {
+        char opt_type[4];
```

Comment:
Does the adam is the only case which loads variables for optimizer?
If it is not, then I think the previous way is more suitable for general cases.